include/envoy/server/process_context.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+
+/**
+ * Represents some other part of the process.
+ */
+class ProcessObject {
+public:
+  virtual ~ProcessObject() {}
+};
+
+/**
+ * Context passed to filters to access resources from non-Envoy parts of the
+ * process.
+ */
+class ProcessContext {
```

Comment:
If we're only ever going to support `get()`, I guess it might be possible to eliminate `ProcessObject` and just pass in a subclass of `ProcessContext` with the bits you are after. That said, I think it's good to have this indirection as we can grow it over time.

One consideration I'm interested in your thoughts on. Let's say we have two filters and they want to get at different aspects of the state. Would it make sense to have a map from a string to the `ProcessObject`s, or have them aware of the internals of `ProcessObject` to do this? I'm thinking the former would make sense when you have multiple parties trying to collaborate on `ProcessObject`, the latter when it's single party.

Reply:
I think it's fine as is, the thing to be aware of then is in the future we might see an API breaking change. Let's ship and iterate.